cmd/main.go

```diff
@@ -149,6 +150,9 @@ func main() {
 		Recorder:         mgr.GetEventRecorderFor("LinodeObjectStorageBucketReconciler"),
 		WatchFilterValue: objectStorageBucketWatchFilter,
 		LinodeApiKey:     linodeToken,
+		LinodeClientFactory: func(apiKey string) scope.LinodeObjectStorageClient {
+			return scope.CreateLinodeClient(apiKey)
+		},
```

Comment:
We can't use a lambda function here. I recommend creating a function in utils/helpers.go and pass it in. Golangci won't be happy